### DISCUSSION TITLE
[mongo-cxx-driver] build failure for feature std-experimental

### DISCUSSION THREAD

**Discussion by prateek9623** [microsoft/vcpkg#18702] [Q&A]
- OS: Windows
- Compiler: Visual c++ 2019

`./vcpkg install mongo-cxx-driver[std-experimental]`

**Failure logs** 
install message:
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:108 (message):
    Command failed: ninja -v
    Working Directory: E:/vcpkg/buildtrees/mongo-cxx-driver/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      E:\vcpkg\buildtrees\mongo-cxx-driver\config-x64-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:319 (vcpkg_execute_required_process)
  ports/mongo-cxx-driver/portfile.cmake:42 (vcpkg_configure_cmake)
  scripts/ports.cmake:128 (include)

config-x64-windows-out.log:
CMake Error at src/bsoncxx/CMakeLists.txt:69 (message):
  Can only use BSONCXX_POLY_USE_STD_EXPERIMENTAL if CMAKE_CXX_STANDARD is 14
  or higher

**Additional context**
Shouldn't vcpkg should set CMAKE_CXX_STANDARD, if not How to set it, as default is 11 in mongo-cxx-driver

**Comment by PhoebeHui** [comment]
@prateek9623, vcpkg doesn't set that variable, you need to create a overlay or custom ports to implement it.
1. You can add set(CMAKE_CXX_STANDARD 11) to mongo-cxx-driver CMakeLists.txt file.
2. Or you can pass the option via VCPKG_CXX_FLAGS in mongo-cxx-driver portfile.cmake file.

**Comment by prateek9623** [comment]
> @prateek9623, vcpkg doesn't set that variable, you need to create a overlay or custom ports to implement it.
> 1. You can add set(CMAKE_CXX_STANDARD 11) to mongo-cxx-driver CMakeLists.txt file.
> 2. Or you can pass the option via VCPKG_CXX_FLAGS in mongo-cxx-driver portfile.cmake file.

actually, I want to set it to 14, what can I do in my manifest file to pass flags

**Comment by PhoebeHui** [comment]
@prateek9623, AFAIK, there is no way to pass the option in manifest file, another workaroud may be set the VCPKG_CXX_FLAGS in toolchain file, however it only works when using vcpkg's built-in toolchains, See https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#prefer-patching-over-overriding-vcpkg_variable-values

**Comment by prateek9623** [comment]
ok so maybe we need to update the port file for this, but is there a way to detect supported standard in port file?

**Comment by PhoebeHui** [comment]
You can try to set VCPKG_CXX_FLAGS in mongo-cxx-driver portfile.cmake file like:

On windows:
set(VCPKG_CXX_FLAGS "${VCPKG_CXX_FLAGS} -std:c++14")

On Unix:
set(VCPKG_CXX_FLAGS "${VCPKG_CXX_FLAGS} -std=c++14")

**Comment by NancyLi1013** [comment]
@prateek9623 

Is this still a problem for you now?

**Comment by prateek9623** [comment]
@NancyLi1013 issue is still there, but I am using boost now
